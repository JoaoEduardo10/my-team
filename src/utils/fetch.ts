/* eslint-disable @typescript-eslint/no-explicit-any */

import { TCountrys, TLeagues, TTeam } from '../types';

const url = import.meta.env.VITE_URL_API as string;

const authUser = async (key: string) => {
  const response = await fetch(`${url}/status`, {
    method: 'GET',
    headers: {
      'x-rapidapi-key': key,
      'X-RapidAPI-Host': 'v3.football.api-sports.io'
    },
    redirect: 'follow'
  });
  const data = await response.json();

  return data;

};

const getCountry = async (key: string) => {
  const response = await fetch(`${url}/countries`, {
    method: 'GET',
    headers: {
      'x-rapidapi-key': key,
      'X-RapidAPI-Host': 'v3.football.api-sports.io'
    },
    redirect: 'follow'
  });
  const data: TCountrys = await response.json();

  return data;
};

const getLeague= async (key: string, country: string, season: number) => {
  const response = await fetch(`${url}/leagues?country=${country}&season=${season}`, {
    method: 'GET',
    headers: {
      'x-rapidapi-key': key,
      'X-RapidAPI-Host': 'v3.football.api-sports.io'
    },
    redirect: 'follow'
  });
  const data: TLeagues = await response.json();

  return data;
};

const getSeason = async (key: string) => {
  const response = await fetch(`${url}/leagues/seasons`, {
    method: 'GET',
    headers: {
      'x-rapidapi-key': key,
      'X-RapidAPI-Host': 'v3.football.api-sports.io'
    },
    redirect: 'follow'
  });
  const data: { response: number[] } = await response.json();

  return data;
};

const getTeam = async (key: string, season: number, league: number) => {
  const response = await fetch(`${url}/teams?league=${league}&season=${season}`, {
    method: 'GET',
    headers: {
      'x-rapidapi-key': key,
      'X-RapidAPI-Host': 'v3.football.api-sports.io'
    },
    redirect: 'follow'
  });
  const data: TTeam = await response.json();

  return data;
};


export { authUser, getCountry, getLeague, getSeason, getTeam };
