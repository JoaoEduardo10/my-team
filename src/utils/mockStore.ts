import configureStore from 'redux-mock-store';


const mockStore = configureStore([]);

const store = mockStore({
  team: {
    country: 'brasil',
    season: '2022',
    league: '47',
    team: 'SÃ£o paulo'
  }
});

export { store };
