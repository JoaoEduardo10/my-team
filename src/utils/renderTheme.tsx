import { render } from '@testing-library/react';
import { ThemeProvider } from 'styled-components';
import { theme } from '../styles/theme';
import { GlobalStyles } from '../styles/globals-styles';
import { RouterProvider, createBrowserRouter } from 'react-router-dom';
import { AuthProvider } from '../context/auth';
import { Provider } from 'react-redux';
import { store } from './mockStore';

const mockRouet = (children: React.ReactNode, path?: string) => {
  return createBrowserRouter([
    {
      path: path ?? '/',
      element: children
    }
  ]);
};

export const renderTheme = (
  children: React.ReactNode,
  path?: string
) => {
  return render(
    <Provider store={store}>
      <ThemeProvider theme={theme}>
        <AuthProvider>
          <RouterProvider router={mockRouet(children, path)} />
        </AuthProvider>
        <GlobalStyles />
      </ThemeProvider>
    </Provider>
  );
};
