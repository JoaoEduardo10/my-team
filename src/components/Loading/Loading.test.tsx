import { screen } from '@testing-library/react';
import { Loading } from '.';
import { renderTheme } from '../../utils/renderTheme';

describe('<Loading />', () => {
  it('should a render Loading', () => {
    renderTheme(<Loading />);

    const loading = screen.getByLabelText('Loading');

    expect(loading).toBeInTheDocument();
    expect(loading).toMatchSnapshot();
  });
});
