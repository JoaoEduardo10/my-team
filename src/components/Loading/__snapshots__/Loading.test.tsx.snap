// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`<Loading /> > should a render Loading 1`] = `
<div
  aria-label="Loading"
  class="hnNaLz"
>
  <div
    class="jJUzso"
  >
    <div
      class="iuwazv"
    />
  </div>
</div>
`;
